components/camel-weka/src/main/java/org/apache/camel/component/weka/WekaEndpoint.java

```diff
@@ -44,14 +68,18 @@ public WekaEndpoint(String uri, WekaComponent component, WekaConfiguration confi
         this.configuration = config;
     }
 
+    public WekaConfiguration getConfiguration() {
+        return configuration;
+    }
+
     @Override
     public WekaComponent getComponent() {
-        return (WekaComponent)super.getComponent();
+        return (WekaComponent) super.getComponent();
     }
 
     @Override
-    public Consumer createConsumer(Processor processor) throws Exception {
-        throw new UnsupportedOperationException();
+    public Consumer createConsumer(Processor processor) {
+        return new WekaConsumer(this, processor);
```

Comment:
You need to call configureConsumer too, see the other components

Reply:
Done